src/ert/scheduler/job.py

```diff
@@ -165,7 +166,58 @@ async def _max_runtime_task(self) -> None:
         )
         self.returncode.cancel()
 
+    async def check_manifest(self, timeout: int = 120) -> None:
+        if self._scheduler.manifest is None:
+            return
+        runpath = self.real.run_arg.runpath
+        keep_waiting = True
+        manifest = self._scheduler.manifest.get(runpath)
+
+        while keep_waiting:
+            keep_waiting = False
+            if manifest is None:
+                timeout -= 1
+                await asyncio.sleep(1)
+                if runpath in self._scheduler.manifest:
+                    manifest = self._scheduler.manifest.get(runpath)
+                    if manifest is None:
+                        break
+                keep_waiting = timeout > 0
+        if not isinstance(manifest, dict):
+            return
+
+        keep_waiting = True
+        while keep_waiting:
+            keep_waiting = False
+            file_paths = []
+            try:
+                file_paths = [info["path"] for _, info in manifest.items()]
+            except (LookupError, AttributeError) as e:
+                logger.exception(f"Invalid manifest exception {e}")
```

Comment:
LookupError and AttributeError should never happen. Instead of logging it and silently returning, I suggest we instead have robust handling of any exception thrown by `_check_manifest`.